### PR TITLE
Add use load flag + isLoading status

### PR DESCRIPTION
**Pull request by timarney**
# Summary | Résumé

- Updates `swr` to latest package version
- Updates useFlag to useLoadFlag which allows us to check the loading state for the API call

Prior to adding the isLoading variable the pages would briefly show the content as if the flag was off i.e. `Registration is closed at this time`.

With the updated flag check we can show a spinner prior to getting the status back.

| Before                                          | After                                        |
| ----------------------------------------------- | -------------------------------------------- |
| https://user-images.githubusercontent.com/62242/211567350-05a3b08a-b012-46e9-ba65-6423dcdadfde.mp4 | https://user-images.githubusercontent.com/62242/211567421-3ea5fc06-6e7f-4c93-846a-467c89051c8e.mp4 |

# Test instructions | Instructions pour tester la modification

- Visit `/signup/register` --- watch the loading progress

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

@bryan-robitaille if we're happy with this approached I will update the other files that are using `useFlag`.  I also opted not to pass the suspense `true` option for now ... some interesting discussion here: https://github.com/vercel/swr/issues/1906

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [ ] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: lib/hooks/useLoadFlag.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import useSWR from "swr";
+import axios from "axios";
+
+const fetcher = (url: string) => axios.get(url).then((response) => response.data);
+
+export const useLoadFlag = (key: string): { isLoading: boolean; status: boolean | undefined } => {
```
Comment (timarney) [inline lib/hooks/useLoadFlag.ts:6]: Note we can update useFlag to use this code but wanted to verify the approach first.

**Comment by bryan-robitaille** [comment]
👍 I love the approach.

**Review by craigzour — APPROVED** [review]
Nice work :)

**Comment by timarney** [comment]
Will merge this and create a follow-up PR to update the remaining files that `useFlag`

**Pull request merged**